packages/libssp.rb

```diff
@@ -3,41 +3,92 @@
-     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libssp/10.3.0_armv7l/libssp-10.3.0-chromeos-armv7l.tar.xz',
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libssp/10.3.0_i686/libssp-10.3.0-chromeos-i686.tar.xz',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libssp/10.3.0_x86_64/libssp-10.3.0-chromeos-x86_64.tar.xz'
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libssp/11.1.0_armv7l/libssp-11.1.0-chromeos-armv7l.tar.xz',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libssp/11.1.0_armv7l/libssp-11.1.0-chromeos-armv7l.tar.xz',
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libssp/11.1.0_i686/libssp-11.1.0-chromeos-i686.tar.xz',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/libssp/11.1.0_x86_64/libssp-11.1.0-chromeos-x86_64.tar.xz'
   })
   binary_sha256({
-    aarch64: '96cccac5b98b142fdfeed83fcd2cc163dfcda386246366b4f7702ef9f1076649',
-     armv7l: '96cccac5b98b142fdfeed83fcd2cc163dfcda386246366b4f7702ef9f1076649',
-       i686: 'a4bf6df2e4e7d325b12c882702ae0a7262fdda7c1a109f7e044a4de7b206bbfc',
-     x86_64: '6c59d395ebd6d461cdc619333566c2139f96209390f7826f58df264f42301c0e'
+    aarch64: '21d6872debb576b442a0a62ab7d0ae4bc765db83bf43cdb8e11c28f56e51d456',
+     armv7l: '21d6872debb576b442a0a62ab7d0ae4bc765db83bf43cdb8e11c28f56e51d456',
+       i686: '5b02bf769ec3754f4b0c937e5420f13bf285bc81bdc9c68fbc6f63c56562682b',
+     x86_64: '356cd568080dd0e724081ec7f020d0b21e091d5552d175746f830c48830bd8b9'
   })
 
+  depends_on 'ccache' => :build
+  depends_on 'dejagnu' => :build # for test
+  depends_on 'glibc' # R
+
   @gcc_name = 'libssp'
-  @gcc_global_opts = '--enable-static \
-                      --enable-shared \
-                      --disable-multilib \
-                      --disable-cet \
-                      --with-pic \
-                      --enable-symvers \
-                      --with-gcc-major-version-only \
-                      --with-system-libunwind'
+
+  @gcc_global_opts = '--disable-libmpx \
+  --disable-libssp \
+  --disable-multilib \
+  --disable-werror \
+  --enable-cet=auto \
+  --enable-checking=release \
+  --enable-clocale=gnu \
+  --enable-default-pie \
+  --enable-default-ssp \
+  --enable-gnu-indirect-function \
+  --enable-gnu-unique-object \
+  --enable-host-shared \
+  --enable-lto \
+  --enable-plugin \
+  --enable-shared \
+  --enable-symvers \
+  --enable-static \
+  --enable-threads=posix \
+  --with-gcc-major-version-only \
+  --with-gmp \
+  --with-isl \
+  --with-mpc \
+  --with-mpfr \
+  --with-pic \
+  --with-system-libunwind \
+  --with-system-zlib'
+
+  @cflags = '-fPIC -pipe'
+  @cxxflags = '-fPIC -pipe'
+  @languages = 'c,c++,jit,objc,fortran,go'
+  case ARCH
+  when 'armv7l', 'aarch64'
+    @archflags = '--with-arch=armv7-a --with-float=hard --with-fpu=neon --with-tune=cortex-a15'
```

Comment:
Should we be assuming all arm systems use the cortex a15 chipset?

Reply:
The a15 is from here: https://www.chromium.org/chromium-os/how-tos-and-troubleshooting/chromiumos-board-porting-guide
```
-mtune=<arch> -- see the GCC manual for complete lists of possible values here
This selects the processor that the code will be optimized best for without requiring it
If you don't specify this, then the -march setting will be used
cortex-a8 or cortex-a9 or cortex-a15 are common for ARM parts
```

(I think _optimized best for without requiring it_ keeps it safe.)

Reply:
I'll remove it from here. Better safe.